chapters/en/Unit 13 - Outlook/RMT.mdx

```diff
@@ -0,0 +1,41 @@
+## Overview
+
+### What is Retention Networks
+RetNet, or Retentive Network, is a foundational architecture proposed for large language models in the paper <paper: RetNet>. This architecture is designed to address key challenges in the realm of large-scale language modeling: Training Parallelism, Low-Cost Inference, and Good performance.
+
+![LLM Challenges.png](https://huggingface.co/datasets/hf-vision/course-assets/blob/main/LLM%20Challenges.png)
+RetNet is able to tackle these challenges by introducing the Multi-Scale Retention (MSR) mechanism, which is an alternative to the multi-head attention mechanism commonly used in Transformer models. 
+MSR has a dual form of recurrence and parallelism, so it is possible to train the models in a parallel way while recurrently conducting inference.
+
+The Multi-Scale Retention mechanism operates under three computation paradigms:
+1) **Parallel Representation**: This aspect of RetNet is designed to fully utilize GPU resources, enabling efficient parallel training of the model.
+![Parallel Representation.png](https://huggingface.co/datasets/hf-vision/course-assets/blob/main/Parallel%20Representation.png)
+
+2) **Recurrent Representation**: This representation facilitates efficient inference with O(1) complexity in terms of memory and computational requirements. It significantly reduces deployment costs and latency, and simplifies implementation by eliminating the need for key-value cache strategies often used in traditional models.
+![Recurrent Representation.png](https://huggingface.co/datasets/hf-vision/course-assets/blob/main/Recurrent%20Representation.png)
+
+3) **Chunkwise Recurrent Representation**: This third paradigm addresses the challenge of long-sequence modeling. It achieves this by encoding each local block in parallel for computational speed while simultaneously encoding global blocks in a recurrent manner to optimize GPU memory usage.
+
+During the training phase, the approach incorporates both parallel and chunkwise recurrent representations, optimizing GPU usage for fast computation and being particularly effective for long sequences in terms of computational efficiency and memory use. 
+For the inference phase, the recurrent representation is used, favoring autoregressive decoding. This method efficiently reduces memory usage and latency, maintaining equivalent performance outcomes.
+
+### From Language to Image
+Paper "Retentive Networks Meet Vision Transformers" <paper: RMT> proposes a new vision backbone inspired by the RetNet architecture. The authors propose RMT to enhance the Vision Transformer (ViT) by introducing explicit spatial priors and reducing computational complexity, drawing inspiration from the RetNet's parallel representation. 
```

Comment:
Same comment here about "<paper: RMT>"

Reply:
Really sorry about this, forgot to attach the link. Working on that.